### PR TITLE
Use gRPC-based liveness probe instead of tetra status

### PR DESCRIPTION
**Pull request by tpapagian**
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://tetragon.io/docs/contribution-guide/submitting-a-pull-request/).
- [x] All code is covered by unit and/or end-to-end tests tests where feasible.
- [x] All commits contain a well written commit message including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Tetragon? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md) in the Cilium repository. 
- [x] Thanks for contributing!

Now, we use tetra status command to report the status of tetragon agent. This comes with some overheads as tetra binary has a lot of additional functionality and it seems like an overkill to use that for status reporting.

On the other hand, k8s supports liveness probes by using an gRPC-based server. This first patch creates a new gRPC health server to report agent status. The second patch changes the helm chart to make use of that.

```release-note
Use gRPC-based liveness probe instead of tetra status.
```

### CODE REVIEW

File: cmd/tetragon/main.go [hunk 1]
```diff
@@ -440,6 +443,51 @@ func tetragonExecute() error {
...
+	healthServer.SetServingStatus("liveness", grpc_health_v1.HealthCheckResponse_SERVING)
+
+	// Create a new gRPC server for health checks and register the healthServer.
+	grpcHealthServer := grpc.NewServer()
+	grpc_health_v1.RegisterHealthServer(grpcHealthServer, healthServer)
+
+	// Start the gRPC server for the health checks.
+	go func() {
+		// the gRPC server for the health checks listens on port 6789
+		listener, err := net.Listen("tcp", ":6789")
```
Comment (michi-covalent) [inline cmd/tetragon/main.go:458]: probably better to make the port configurable 💭

File: cmd/tetragon/main.go [hunk 2]
```diff
@@ -440,6 +443,51 @@ func tetragonExecute() error {
...
+
+		if err = grpcHealthServer.Serve(listener); err != nil {
+			log.WithError(err).Fatal("Failed to start gRPC healthserver")
+		}
+	}()
+
+	// Every 10 seconds check the agent health. To check if our agent is health we call
+	// health.GetHealth() and we report the status to the healthServer.
+	go func() {
+		ticker := time.NewTicker(10 * time.Second)
```
Comment (michi-covalent) [inline cmd/tetragon/main.go:471]: ... also might be good to make the health check interval configurable.

File: install/kubernetes/tetragon/values.yaml [hunk 3]
```diff
@@ -64,9 +64,11 @@ tetragon:
...
     privileged: true
   # -- Overrides the default livenessProbe for the tetragon container.
-  livenessProbe: {}
-  #  grpc:
-  #    port: 54321
+  livenessProbe:
+    timeoutSeconds: 60
+    grpc:
+      port: 6789
+      service: "liveness"
```
Comment (michi-covalent) [inline install/kubernetes/tetragon/values.yaml:71]: maybe keep this value as is and update https://github.com/cilium/tetragon/blob/4c47837a06f473b45ad189149c63853f2f502304/install/kubernetes/tetragon/templates/_container_tetragon.tpl#L70-L81 instead just to follow the same pattern as other values like these => https://github.com/cilium/tetragon/blob/4c47837a06f473b45ad189149c63853f2f502304/install/kubernetes/tetragon/values.yaml#L57-L63)

**Review by michi-covalent — COMMENTED** [review]
@lambdanis have we already started writing version specific upgrade guide? it might make sense to document this change, something like:

```
In v1.2 tetragon container uses the grpc liveness probe by default. To continue using "tetra status" for liveness probe,
specify tetragon.livenessProbe Helm value. For example:

tetragon:
  livenessProbe:
     timeoutSeconds: 60
     exec:
       command:
       - tetra
       - status
       - --server-address
       - "54321"
       - --retries
       - "5"
```

File: cmd/tetragon/main.go (hunk 1 above)
Comment (tpapagian) [reply cmd/tetragon/main.go]: Done. I have also added a way to disable the gRPC server if needed. All of these are now also configurable through configmap.

File: cmd/tetragon/main.go (hunk 2 above)
Comment (tpapagian) [reply cmd/tetragon/main.go]: Done!

File: install/kubernetes/tetragon/values.yaml (hunk 3 above)
Comment (tpapagian) [reply install/kubernetes/tetragon/values.yaml]: Makes sense. I have also done this update.

File: install/kubernetes/tetragon/templates/_container_tetragon.tpl [hunk 4]
```diff
@@ -70,14 +70,9 @@
 {{- else if .Values.tetragon.grpc.enabled }}
```
Comment (lambdanis) [inline install/kubernetes/tetragon/templates/_container_tetragon.tpl:70]: I think it should check `{{- .Values.tetragon.healthGrpc.enabled }}`, right?

**Review by lambdanis — COMMENTED** [review]
One Helm detail, but looks good, thanks!

File: install/kubernetes/tetragon/templates/_container_tetragon.tpl (hunk 4 above)
Comment (tpapagian) [reply install/kubernetes/tetragon/templates/_container_tetragon.tpl]: Thanks, fixed!

**Review by kkourt — APPROVED** [review]

**Review by lambdanis — APPROVED** [review]

**Pull request merged**